### DISCUSSION TITLE
route error

### DISCUSSION THREAD

**Discussion by albaloo2000** [firefly-iii/firefly-iii#6340] [Firefly III]
Anyone else getting this error? 

Running php artisan route:list shows the following: 

   Illuminate\Contracts\Container\BindingResolutionException 

  Target class [FireflyIII\Api\V1\Controllers\Models\Recurrence\RecurrenceController] does not exist.

I have two different installations and both showing the same error.  One is on a otherwise working server.

**Comment by albaloo2000** [comment]
JC let me know if this is all your work in progress. 

Under routes/api.php,  the final line refers to RecurrenceController but the name space is Models where no such controller exists. 



Route::group(
    ['namespace' => 'FireflyIII\Api\V1\Controllers\Models\Recurrence', 'prefix' => 'recurrences',
     'as'        => 'api.v1.recurrences.',],
    static function () {

        Route::get('', ['uses' => 'ShowController@index', 'as' => 'index']);
        Route::post('', ['uses' => 'StoreController@store', 'as' => 'store']);
        Route::get('{recurrence}', ['uses' => 'ShowController@show', 'as' => 'show']);
        Route::put('{recurrence}', ['uses' => 'UpdateController@update', 'as' => 'update']);
        Route::delete('{recurrence}', ['uses' => 'DestroyController@destroy', 'as' => 'delete']);

        Route::get('{recurrence}/transactions', ['uses' => 'ListController@transactions', 'as' => 'transactions']);
// See reference nr. 16
        Route::post('trigger', ['uses' => 'RecurrenceController@trigger', 'as' => 'trigger']);
    }
);

**Comment by albaloo2000** [comment]
I assume this is all from work in progress.   I had to fix about 5 different references but routes:list is now working.

**Comment by JC5** [comment]
I've never used that command, so I'm not surprised it didn't work. I hope you found it interesting!

**Comment by albaloo2000** [comment]
It shows a few of your routes are broken.  They are mostly pointing to controllers that don't exist.

**Reply by JC5** [reply]
Thanks, That's ok, I don't use them anywhere.
